packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx

```diff
@@ -53,7 +54,7 @@ export type QueryContextType<
   /** Returns a deep clone of the initial query bundle request */
   getInitQueryRequest: () => QueryBundleRequest
   /** Updates the current query with the passed request */
-  executeQueryRequest: (param: QueryBundleRequest) => void
+  executeQueryRequest: (param: React.SetStateAction<QueryBundleRequest>) => void
```

Comment:
Allow passing an updater function (this just uses a `useState` setter, so this was always supported)